app/src/main/java/ru/yandex/practicum/contacts/presentation/base/BaseListDiffCallback.java

```diff
@@ -4,21 +4,45 @@
 import androidx.annotation.Nullable;
 import androidx.recyclerview.widget.DiffUtil;
 
-public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T>{
+public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> implements ListDiffInterface<T>{
```

Comment:
```suggestion
public class BaseListDiffCallback<T extends ListDiffInterface<T>> extends DiffUtil.ItemCallback<T> {
```
Сам класс не должен этот интерфейс наследовать. Интерфейс сделан для классов, которые представляют собой элементы списка, чтобы можно было удобно их сравнивать друг между другом

Reply:
> Сам класс не должен этот интерфейс наследовать. Интерфейс сделан для классов, которые представляют собой элементы списка, чтобы можно было удобно их сравнивать друг между другом

в задании было указано, сделайте его обязательным, подумал, почему бы не сделать так, но кажется примерно начал понимать, почему не так